spec/models/post_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it { should belong_to(:author).class_name('User') }
+  it { should have_many(:comments) }
+  it { should have_many(:likes) }
+
+  # Test callbacks
+  describe 'after_save' do
+    let(:post) { create(:post) }
+    let(:author) { post.author }
+
+    it 'should call update_posts_counter' do
+      expect(post).to receive(:update_posts_counter)
+      post.save
+    end
+
+    it 'should increment posts_counter of the author' do
+      expect { post.save }.to change { author.reload.posts_counter }.by(1)
+    end
+  end
+
+  # Test validations
+  it { should validate_presence_of(:title) }
+  it { should validate_length_of(:title).is_at_most(250) }
+  it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+  it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
```

Comment:
- [optional] Hello, kudos working on the changes required earlier; please kindly take note that your tests are failing at the moment and in the previous review. Kindly take your time to ensure that your are consistent with your variable names most especially (posts_counter) for users table. More so, ensure that all the methods are well defined. Below is the screenshot that I got 👇 

<img width="613" alt="Untitled" src="https://github.com/AlaaAlsalem/blogaapp/assets/103386183/25a99a7b-2203-454a-b16c-6b000889bbff">

**I marked it as optional because the previous reviewer skipped it but I implore you strongly to implement it, because I will try and follow this PR up** Thank you.